src/events/google_cloud.js

```diff
@@ -99,14 +100,61 @@ const getMessageData = (reqOptsMessages, index) => {
         reqOptsMessages.length > index &&
         reqOptsMessages[index].data
     ) {
-        const messageData = JSON.parse(`${reqOptsMessages[index].data}`);
-        if (typeof messageData === 'object') {
-            return messageData;
+        try {
+            const messageData = JSON.parse(`${reqOptsMessages[index].data}`);
+            if (typeof messageData === 'object') {
+                return messageData;
+            }
+        } catch (err) {
+            return null;
         }
     }
     return null;
 };
 
+const handlePublishMethod = (messages, config) => {
+    const messageIds = messages && messages.messageIds ? messages.messageIds : messages;
+    if (Array.isArray(messageIds)) {
+        const reqOptsMessages = config.reqOpts && config.reqOpts.messages;
+        const parsedMessages = messageIds.reduce((acc, messageId, currentIndex) => {
+            let message = { id: messageId };
+            // add message only when METADATA_ONLY === FALSE
+            if (!epsagonConfig.getConfig().metadataOnly) {
```

Comment:
use the third argument to `addMetadata` instead

Reply:
I also wanted to do it, But here I adding the message data to an object who must add to the event.